diff --git a/libiec61850/CMakeLists.txt b/libiec61850/CMakeLists.txt
index 50c7a7c7..ef46bf28 100644
--- a/libiec61850/CMakeLists.txt
+++ b/libiec61850/CMakeLists.txt
@@ -28,8 +28,8 @@ test_big_endian(PLATFORM_IS_BIGENDIAN)
 set(CONFIG_MMS_MAXIMUM_PDU_SIZE "65000" CACHE STRING "Configure the maximum size of an MMS PDU (default 65000)"   )
 set(CONFIG_MAXIMUM_TCP_CLIENT_CONNECTIONS 5 CACHE STRING "Configure the maximum number of clients allowed to connect to the server")
 
-option(BUILD_EXAMPLES "Build the examples" ON)
-option(BUILD_PYTHON_BINDINGS "Build Python bindings" OFF)
+option(BUILD_EXAMPLES "Build the examples" OFF)
+option(BUILD_PYTHON_BINDINGS "Build Python bindings" ON)
 
 option(CONFIG_MMS_SINGLE_THREADED "Compile for single threaded version" ON)
 option(CONFIG_MMS_THREADLESS_STACK "Optimize stack for threadless operation (warning: single- or multi-threaded server will not work!)" OFF)
