diff --git a/libiec61850/pyiec61850/iec61850.i b/libiec61850/pyiec61850/iec61850.i
index 26f6655f..8fbbd942 100644
--- a/libiec61850/pyiec61850/iec61850.i
+++ b/libiec61850/pyiec61850/iec61850.i
@@ -104,7 +104,7 @@ void GooseSubscriber_setDstMac(GooseSubscriber subscriber,
 #include "eventHandlers/reportControlBlockHandler.hpp"
 #include "eventHandlers/gooseHandler.hpp"
 #include "eventHandlers/commandTermHandler.hpp"
-std::map< std::string, EventSubscriber*> EventSubscriber::m_subscriber_map = {};
+std::map< std::string, EventSubscriber*> EventSubscriber::m_subscriber_map;
 %}
 %include "eventHandlers/eventHandler.hpp"
 %include "eventHandlers/reportControlBlockHandler.hpp"
@@ -145,3 +145,15 @@ void CommParameters_setDstAddress(CommParameters *gooseCommParameters,
                                   uint8_t dst_mac_4,
                                   uint8_t dst_mac_5);
 
+%{
+#include "callbackWrapper.hpp"
+%}
+
+%pythoncallback;
+ControlHandlerResult ControlHandlerProxy (ControlAction action, void* parameter, MmsValue* ctlVal, bool test);
+void ReportHandlerProxy (void* parameter, ClientReport report);
+%nopythoncallback;
+
+%ignore ControlHandlerProxy;
+%ignore ReportHandlerProxy;
+%include "callbackWrapper.hpp"
